[R2][KV][DO] Comparison table update (#6468)
cloudflare/cloudflare-docs: content/workers/platform/storage-objects.md

{{<table-wrap>}}

| Feature                                       | KV           | R2           | DO                   |
| --------------------------------------------- | ------------ | ------------ |--------------------- |
| Maximum storage per account                   | Unlimited<sup>1</sup>|Unlimited              | 50 GiB               |

> | Grouping name                                 | Namespace    | Bucket       | Durable Object       |

Author: That's a good point. What is the mental model distinction between namespace and DO that people should have? Namespace = code grouping and DO = data grouping?

> | Grouping name                                 | Namespace    | Bucket       | Durable Object       |

Author: Reworded things a bit and added a footnote as discussed offline.

| Groups per account                            | 100          | 1000         | Unlimited            |
| Maximum keys per grouping                     | Unlimited    | Unlimited    | Unlimited            |

> | Maximum key size                              | 512 bytes    | 1024 bytes   | 2048 bytes           |
> | Maximum metadata per key                      | 1024 bytes   | 2048 bytes   | N/A                  |

Reviewer: ```suggestion
| Maximum key size                              | 512 bytes    | 1,024 bytes   | 2,048 bytes           |
| Maximum metadata per key                      | 1,024 bytes   | 2,048 bytes   | N/A                  |
```

| Consistency model                             | Eventual     | Strong       | Transactional for multiple keys in a single DO |
| Cached                                        | Always       | Programatically using the [Worker Cache API](/workers/runtime-apis/cache/) or configure a custom [public bucket](/r2/data-access/public-buckets) domain. | Possible when using [Cache API](/workers/runtime-apis/cache/) |
| S3-compatible API                             | No           | Yes          | No                   |
| TTL expiration                                | Object-level | Not currently available | Not automatic, but possible using [alarms](/workers/learning/using-durable-objects/#alarms-in-durable-objects) |
| Maximum operations per second                 | Unlimited cached reads |  10,000+ reads/s per bucket, 1,000+ writes/s per bucket<sup>2</sup> | 100+ requests/second per DO<sup>2</sup> |

{{</table-wrap>}}

<sup>1</sup>Free accounts are limited to 1GiB.
<sup>2</sup>Performance may depend on the specific data access patterns of your application and may be lower or higher depending on your specific application.

> This is a comparison of the baseline available across customers. To increase these limits,
> contact your Cloudflare account team.

Reviewer: ```suggestion
```

> This is a comparison of the baseline available across customers. To increase these limits,
> contact your Cloudflare account team.

Reviewer: ```suggestion
Some of the above limits can be increased. For more information, contact your Cloudflare account team.
```

| Cached                                        | Always       | Programatically using the [Worker Cache API](/workers/runtime-apis/cache/) or configure a custom [public bucket](/r2/data-access/public-buckets) domain. | Possible when using [Cache API](/workers/runtime-apis/cache/) |
| S3-compatible API                             | No           | Yes          | No                   |
| TTL expiration                                | Object-level | Not currently available | Not automatic, but possible using [alarms](/workers/learning/using-durable-objects/#alarms-in-durable-objects) |
| Maximum operations per second                 | Unlimited cached reads |  10,000+ reads/s per bucket, 1,000+ writes/s per bucket<sup>3</sup> | 150 requests/second per DO<sup>3</sup> |

{{</table-wrap>}}

<sup>1</sup>Free accounts are limited to 1GiB.
<sup>2</sup>A Durable Object namespace is a logical container for as many Durable Objects as you need and is backed by a class implementing the logic all those Durable Objects will share.
<sup>3</sup>Performance may depend on the specific data access patterns of your application and may be lower or higher depending on your specific application.

> This is a comparison of the baseline limits available across customers.
> We want to encourage you to build any application you can dream up, and realize that doesn't always fit within our limits.
> 
> To increase a limit, please fill out the [Limit Increase Request form](https://forms.gle/ukpeZVLWLnKeixDu7).

Reviewer: ```suggestion
You can request adjustments to limits that conflict with your project goals by contacting Cloudflare. To increase a limit, complete the [Limit Increase Request form](https://forms.gle/ukpeZVLWLnKeixDu7).
```
